plugins/ipam/ipstore/ipstore.go

```diff
@@ -159,6 +173,53 @@ func (manager *IPManager) Release(ip string) error {
 	return nil
 }
 
+// ReleaseByID release the key-value pair by id
+func (manager *IPManager) ReleaseByID(id string) (string, error) {
+	// TODO improve this part by implement listing all the kv pairs
+	// libkv library only provide list all key-value pairs with prefix of key, and
+	// if the result is empty it will return store.ErrKeyNotFound error
+	kvPairs, err := manager.client.List(IPPrefix)
+	if err == store.ErrKeyNotFound {
+		return "", errors.Errorf("release ipstore: no ip associated with the id: %s", id)
+	}
+	if err != nil {
+		return "", errors.Wrapf(err, "release ipstore: failed to list the key-value pairs in db")
+	}
+
+	var ipDeleted string
+	for _, kvPair := range kvPairs {
+		if string(kvPair.Value) == id {
+			err = manager.Release(kvPair.Key)
+			if err != nil {
+				return "", err
+			}
+			ipDeleted = kvPair.Key
+			return ipDeleted, nil
+		}
+	}
+	return "", errors.Errorf("release ipstore: no ip address associated with the given id: %s", id)
+}
+
+// UniqueID checks whether the id has already existed in the ipam
+func (manager *IPManager) UniqueID(id string) (bool, error) {
+	kvPairs, err := manager.client.List(IPPrefix)
+	// TODO improve this part by implement listing all the kv pairs
+	if err == store.ErrKeyNotFound {
+		return true, nil
+	}
+
+	if err != nil {
+		return false, errors.Wrapf(err, "release ipstore: failed to list the key-value pairs in db")
```

Comment:
I don't think you need `release ` here. Just `ipstore: failed..` should be sufficient